### ISSUE TITLE
[BUG] TinyTimeMixer should not create validation set if validation_split is None

### ISSUE THREAD

**Issue by geetu040** [sktime/sktime#7644]
**Describe the bug**

`TinyTimeMixerForecaster` always creates a validation set even if `validation_split is None`, in which case the `temporal_train_test_split` defaults the `validation_split` to `0.25` 

Following fixes should be made:
- create `y_test` only if `validation_split is not None`
- rename data split: `test` to `eval`

**To Reproduce**

```python
from sktime.forecasting.ttm import TinyTimeMixerForecaster
from sktime.datasets import load_airline
y = load_airline()

forecaster = TinyTimeMixerForecaster(
	config={"context_length": 10},
	training_args={
		"output_dir": "test_output",
		"report_to": "none",
	},
	validation_split=None, # defaults validation_split to 0.25
)
forecaster.fit(y, fh=[1, 2, 3])
y_pred = forecaster.predict()
```

**Expected behavior**

when `validation_split is None`, `test` should not be created and not passed to the `Trainer`

**Versions**
<details>

```
System:
    python: 3.12.5 | packaged by Anaconda, Inc. | (main, Sep 12 2024, 18:27:27) [GCC 11.2.0]
executable: /home/geetu/miniconda3/envs/ai/bin/python
   machine: Linux-6.8.0-51-generic-x86_64-with-glibc2.39

Python dependencies:
          pip: 24.2
       sktime: 0.34.0
      sklearn: 1.5.2
       skbase: 0.8.3
        numpy: 1.26.4
        scipy: 1.14.1
       pandas: 2.2.3
   matplotlib: 3.9.2
       joblib: 1.4.2
        numba: None
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: 2.4.1
   tensorflow: 2.18.0
```

</details>

**Comment by cheachu** [comment]
hey can I try this out?

**Comment by geetu040** [comment]
@cheachu sure you can try this
Check this [contributing guide](https://www.sktime.net/en/stable/developer_guide/git_workflow.html) and let me know if you have any questions.
Good Luck!

**Comment by cheachu** [comment]
@Sohaib-Ahmed21 ,
ok I'll go through it 
Thanks!

**Comment by geetu040** [comment]
> `validation_split` logic in https://github.com/sktime/sktime/blob/main/sktime/forecasting/hf_transformers_forecaster.py might help you out.

actually the condition `validation_split is None` sounds more reasonable than `validation_split == 0`, we should use that instead. I'll update the issue title and message.

**Comment by RHYTHM2405** [comment]
@geetu040 I have changed the trainer logic , adding an if else condition for this.

**Comment by geetu040** [comment]
@cheachu are you working on this? If so, can you please create a draft pull request? I will prioritize your work since you came here first

**Comment by JyotiP24** [comment]
Hi, can I try this?

**Comment by geetu040** [comment]
> Hi, can I try this?

@JyotiP24 thanks for showing interest, but this is already been worked here https://github.com/sktime/sktime/pull/7647
You can review or make suggestions there

**Comment by cheachu** [comment]
> [@cheachu](https://github.com/cheachu) are you working on this? If so, can you please create a draft pull request? I will prioritize your work since you came here first

hey @geetu040 

Thanks for prioritization :)
Was unavailable for the past 2 days and looks like @RHYTHM2405  has a good fix 
I guess he can continue with it.
